### PR TITLE
III-3710 Internalize the udb3-models dependency

### PR DESCRIPTION
**Pull request by LucWollants**
### Changed

- Moved the ud3-models dependency into udb3-silex

---
Ticket: https://jira.uitdatabank.be/browse/III-3710

### CODE REVIEW

File: src/Model/Place/ImmutablePlace.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+     * @param TranslatedTitle $title
+     * @param TranslatedAddress $address
+     * @return ImmutablePlace
+     */
+    public static function createDummyLocation(
+        Language $mainLanguage,
+        TranslatedTitle $title,
+        TranslatedAddress $address
+    ) {
+        /** @phpstan-ignore-next-line */
```
Comment (bertramakers) [inline src/Model/Place/ImmutablePlace.php:125]: Nit: I think this one is easy to fix by making the class and/or constructor private

File: src/Model/ValueObject/Contact/BookingAvailability.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class BookingAvailability extends DateTimeImmutableRange
+{
+    /**
+     * @param \DateTimeImmutable $from
+     * @return static
+     */
+    public static function from(\DateTimeImmutable $from)
+    {
+        /** @phpstan-ignore-next-line */
```
Comment (bertramakers) [inline src/Model/ValueObject/Contact/BookingAvailability.php:15]: Nit: Same as `ImmutablePlace`, making the class `final` fixes this I think

File: src/Model/ValueObject/Contact/BookingAvailability.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return new static($from, null);
+    }
+
+    /**
+     * @param \DateTimeImmutable $to
+     * @return static
+     */
+    public static function to(\DateTimeImmutable $to)
+    {
+        /** @phpstan-ignore-next-line */
```
Comment (bertramakers) [inline src/Model/ValueObject/Contact/BookingAvailability.php:25]: Same nit as above

File: src/Model/ValueObject/Contact/BookingAvailability.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * @param \DateTimeImmutable $from
+     * @param \DateTimeImmutable $to
+     * @return static
+     */
+    public static function fromTo(\DateTimeImmutable $from, \DateTimeImmutable $to)
+    {
+        /** @phpstan-ignore-next-line */
```
Comment (bertramakers) [inline src/Model/ValueObject/Contact/BookingAvailability.php:36]: Again same nit as above 😄

**Review by bertramakers — APPROVED** [review]
Looks good! Some of the ignored PHPStan errors can easily be fixed I think. Only the `Collection` is harder.

File: src/Model/Place/ImmutablePlace.php (hunk 1 above)
Comment (LucWollants) [reply src/Model/Place/ImmutablePlace.php:125]: You are right! I was so focused on the error inside `Collection` I forgot the "easy" fix to make the class `final`.
I will check what can be fixed with the `final` keyword.

**Pull request merged**